samples/mpu/mpu_stack_guard_test/src/main.c

```diff
@@ -10,62 +10,80 @@
 #include <misc/printk.h>
 
 /* size of stack area used by each thread */
-#define STACKSIZE 1024
+#define STACKSIZE  512
 
 /* scheduling priority used by each thread */
 #define PRIORITY 7
 
 /* delay between greetings (in ms) */
 #define SLEEPTIME 500
 
+/* delay between greetings (in ms) */
+#define SCHEDTIME 30
+
 /* Focaccia tastes better than DEEDBEEF ;-) */
 #define STACK_GUARD_CANARY 0xF0CACC1A
 
 struct stack_guard_buffer {
 	/* Make sure canary is not optimized by the compiler */
+	struct k_thread thread;
 	volatile u32_t canary;
 	K_THREAD_STACK_MEMBER(stack, STACKSIZE);
 };
 
-struct stack_guard_buffer buf;
-struct k_thread test_thread;
+#define LAST_THREAD_NUM 3
+struct stack_guard_buffer buf[LAST_THREAD_NUM+1];
 
-u32_t recursive_loop(u32_t counter)
+u32_t recursive_loop(u32_t counter, int num, void *dummy)
 {
-	if (buf.canary != STACK_GUARD_CANARY) {
-		printk("Canary = 0x%08x\tTest not passed.\n", buf.canary);
+
+	if (buf[num].canary != STACK_GUARD_CANARY) {
+		printk("Canary = 0x%08x\tTest not passed.\n", buf[num].canary);
 
 		while (1) {
 			k_sleep(SLEEPTIME);
 		}
 	}
-
+	k_sleep(SCHEDTIME);
```

Comment:
After looking at this more, it doesn't make any difference having it or not.  So I am going to remove it.